### PR TITLE
Issue #13423 Fix - Headcrab death check

### LINKED ISSUE #13423: Dead headcrabs consume adjacent dead simple mobs
**Issue Description**:
When placed next to a dead simple mob of apparently any kind, a headcrab will consume it with the message "The headcrab consumes whole!" _even when dead._

**What did you expect to happen**: 
The headcrabs to not eat things in death.

**What happened instead**:
They did.

**Why is this bad/What are the consequences**:
Ian might get ate for a dumb reason.

**Steps to reproduce the problem**:
Place dead simple mob next to dead headcrab.

**When did the problem start happening**:
Probably since headcrabs became a thing.

**Extra information**:
The dead headcrab does not interact with dead carbons or living simple mobs in any way. Just dead simple mobs. I've tried a slime, a fox and a lavawing watcher in this way.

### PR DESCRIPTION
**Pull request by PidgeyThePirate**
## What Does This PR Do
Fixes #13423 - dead headcrabs eating adjacent dead animals.

## Why It's Good For The Game
Unintentional behaviour is bad, mkay?

## Changelog
:cl:
fix: dead headcrabs no longer consume adjacent dead animals.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/headcrab.dm [hunk 1]
```diff
@@ -25,7 +25,9 @@
 	var/list/human_overlays = list()
 
 /mob/living/simple_animal/hostile/headcrab/Life(seconds, times_fired)
-	if(!is_zombie && isturf(src.loc) && stat != DEAD)
+	if(stat == DEAD)
+		return
```
Comment (Kyep) [inline code/modules/mob/living/simple_animal/hostile/headcrab.dm:29]: Won't this cause ..() at the end to never run?
Won't that cause issues with any other code that runs in Life()?

Comment (PidgeyThePirate) [reply code/modules/mob/living/simple_animal/hostile/headcrab.dm:29]: You're right. One sec.

**Comment by PidgeyThePirate** [comment]
Fixed.

File: code/modules/mob/living/simple_animal/hostile/headcrab.dm [hunk 2]
```diff
@@ -25,22 +25,23 @@
 	var/list/human_overlays = list()
 
 /mob/living/simple_animal/hostile/headcrab/Life(seconds, times_fired)
-	if(!is_zombie && isturf(src.loc) && stat != DEAD)
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/hostile/headcrab.dm:28]: just put this under `if(..())`

File: code/modules/mob/living/simple_animal/hostile/headcrab.dm [hunk 3]
```diff
@@ -25,22 +25,23 @@
...
+			for(var/mob/living/simple_animal/K in oview(src, 1)) //Only for corpse right next to/on same tile
+				if(K.stat == DEAD || (!K.check_death_method() && K.health <= HEALTH_THRESHOLD_DEAD))
+					visible_message("<span class='danger'>[src] consumes [target] whole!</span>")
+					if(health < maxHealth)
+						health += 10
+					qdel(K)
+					break
+				cycles = 0
+		cycles++
 	..()
```
Comment (Fox-McCloud) [inline code/modules/mob/living/simple_animal/hostile/headcrab.dm:45]: and get rid of this.

**Pull request merged**